### PR TITLE
Update duckdb version. Closes #569

### LINKED ISSUE #569: Upgrade DuckDB to 0.7
DuckDB 0.7 contains a few features benefit Lance / lance duckdb extension.

* DuckDB python SDK can query lance lazy loading dataset
* DuckDB C API has `ListVector` manipulation routings.

### PR DESCRIPTION
**Pull request by changhiskhan**
Starting v0.7, duckdb supports lazy lance dataset. ListVector is also exposed via C API.

### CODE REVIEW

**Comment by eddyxu** [comment]
Need to do two submodule checkout as well.

File: python/README.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ Why you should use Lance
 pip install pylance
 ```
 
+Make sure you have a recent version of pandas (1.5+), pyarrow (10.0+), and DuckDB (7.0+)
```
Comment (eddyxu) [inline python/README.md:28]: Should we add this to pylance's pyproject.toml as dependency restriction?

**Review by eddyxu — APPROVED** [review]
Pending last few comments. Also could you make sure the duckdb ext actually builds?

File: python/README.md (hunk 1 above)
Comment (changhiskhan) [reply python/README.md]: done

File: integration/duckdb_lance/Cargo.toml [hunk 2]
```diff
@@ -14,7 +14,7 @@ futures = "0.3"
 num-traits = "0.2"
 
 [dev-dependencies]
-libduckdb-sys = { version = "0.6.1", features = ["bundled"] }
+libduckdb-sys = { version = "7.0.0", features = ["bundled"] }
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply integration/duckdb_lance/Cargo.toml]: done

**Pull request merged**